### ISSUE TITLE
kfctl should use the file path passed to -f not app.yaml

### ISSUE THREAD

**Issue by jlewi** [kubeflow/kfctl#54]
Right now kfctl is hardcoding the path it saves the KFDef to as "app.yaml"

This leads to a confusing user experience; especially if a user needs to rerun kfctl

e.g. they would initially run

```
kfctl apply -f kfctl_gcp_iap.yaml
```

But none of the changes made by kfctl are preserved in kfctl_gcp_app.yaml; they are saved to app.yaml

So if a user needs to rerun they need to do

```
kfctl apply -f app.yaml
```

We should fix that. 

Once kfctl is migrated to KFConfig (kubeflow/kubeflow#4250), I'm hoping this is is a straightforward fix. I think we can just add a field to KFConfig to store the appfile that gets set when the file is loaded.

The write methods can then get the path from that field.

**Comment by swiftdiaries** [comment]
If the user has to rerun, they would have to move into the $KFAPP dir and run `kfctl apply`.
Because apply would have already built the KFAPP.

**Comment by jlewi** [comment]
@swiftdiaries The desired syntax is

```
kfctl apply -f ${KFAPP}/${KFDEF_CONFIG_FILE}
```

So the directory from which kfctl apply is run should only really matter in terms of how relative paths are interpreted.

**Comment by jlewi** [comment]
Not specifying the -f flag is deprecated syntax that we at best are going to keep for backwards compatibility but should remove in 1.0.

**Comment by jlewi** [comment]
@gabrielwen @lluunn @swiftdiaries does someone want to tackle this? It looks like #4250 is merged.

**Comment by lluunn** [comment]
how about delete?
Do we allow users to do `kfctl delete` in the appdir, or the users should do `kfctl delete -f xx.yaml`?

**Comment by jlewi** [comment]
Always "-f" even for delete. 

e.g .

```
kfctl delete -f xx.yaml
```

**Comment by jlewi** [comment]
Verifying with
kubeflow/kubeflow@0ba828a18ee97094a5aa24da2fe09ca75d05d15d
kubeflow/manifests@fccc0ac71efa0ca7816e4f9aa59d03d9dc6934d4
 v0.7-branch

Here's the contents of my app directory

```
ls -la
total 32
drwxr-xr-x  5 jlewi primarygroup 4096 Oct 23 20:47 .
drwxr-xr-x 46 jlewi primarygroup 4096 Oct 23 20:47 ..
drwxr-xr-x  2 jlewi primarygroup 4096 Oct 23 20:47 .cache
drwxr-xr-x  2 jlewi primarygroup 4096 Oct 23 20:47 gcp_config
-rw-r--r--  1 jlewi primarygroup 9472 Oct 23 20:47 kfctl_gcp_iap.0.7.0.yaml
drwxr-xr-x 43 jlewi primarygroup 4096 Oct 23 20:47 kustomize

```

So no app.yaml file is created.

Here's a copy of my YAML file. Based on the file we can see it got updated

* Status has repos set
* Various application parameters like IAP headers and my email got set

Note I added the ".txt" extension because github requires that; kfctl didn't add that.

[kfctl_gcp_iap.0.7.0.yaml.txt](https://github.com/kubeflow/kfctl/files/3765640/kfctl_gcp_iap.0.7.0.yaml.txt)

So this looks to be fixed so I'm closing this issue.
